crates/services/relayer/src/service.rs

```diff
@@ -168,15 +213,44 @@ where
         &mut self,
         eth_sync_gap: &state::EthSyncGap,
     ) -> anyhow::Result<()> {
+        let mut successful_rpc_calls: u64 = 0;
+        let mut rpc_error: bool = false;
+
+        println!(
+            "Starting download with page size: {}",
+            self.adaptive_page_sizer.page_size()
+        );
+        println!("Sync gap: {:?}", eth_sync_gap);
+
         let logs = download_logs(
             eth_sync_gap,
             self.config.eth_v2_listening_contracts.clone(),
             &self.eth_node,
-            self.config.log_page_size,
-        );
+            self.adaptive_page_sizer.page_size(),
+        )
+        .inspect(|result| match result {
+            Ok(downloaded) => {
+                successful_rpc_calls = successful_rpc_calls.saturating_add(1);
+                println!(
+                    "Successful RPC call #{}, blocks: {}..{}",
+                    successful_rpc_calls, downloaded.start_height, downloaded.last_height
+                );
+            }
+            Err(_) => {
+                rpc_error = true;
+                println!("RPC error encountered");
+            }
+        });
+
         let logs = logs.take_until(self.shutdown.while_started());
+        let result = write_logs(&mut self.database, logs).await;
+
+        println!("Total successful RPC calls: {}", successful_rpc_calls);
+        println!("Had RPC error: {}", rpc_error);
 
-        write_logs(&mut self.database, logs).await
+        self.adaptive_page_sizer
+            .update(successful_rpc_calls, rpc_error);
```

Comment:
We only will update the page size at the end of long running function. In the case if we need to sync 200k blocks, we want to increase the page size more often.